### PR TITLE
Swift: Add more path injection sinks

### PR DESCRIPTION
**Pull request by atorralba**

### CODE REVIEW

File: swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift [hunk 1]
```diff
@@ -100,65 +142,75 @@ func test() {
...
+    let _ = fm.createSymbolicLink(atPath: "", pathContent: remoteString) // $ hasPathInjection=139
+    let _ = fm.pathContentOfSymbolicLink(atPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.replaceItemAtURL(originalItemURL: remoteNsUrl, withItemAtURL: safeNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=139
+    let _ = fm.replaceItemAtURL(originalItemURL: safeNsUrl, withItemAtURL: remoteNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=139
+
+    NSString().write(to: remoteUrl, atomically: true, encoding: 0)  // $ hasPathInjection=139
+    NSString().write(toFile: remoteString, atomically: true, encoding: 0) // $ hasPathInjection=139
+    let _ = NSKeyedUnarchiver().unarchiveObject(withFile: remoteString) // $ hasPathInjection=139
+    let _ = ArchiveByteStream.fileStream(fd: remoteString as! FileDescriptor, automaticClose: true) // $ hasPathInjection=139
+    ArchiveByteStream.withFileStream(fd: remoteString as! FileDescriptor, automaticClose: true) { _ in } // $ hasPathInjection=139
```
Comment (geoffw0) [inline swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift:209]: Where is the implementation used by `remoteString as! FileDescriptor`?

**Review by geoffw0 — APPROVED** [review]
Looks good to me.  One small question.

File: swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift (hunk 1 above)
Comment (atorralba) [reply swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift:209]: Do you mean the functions being called? They're stubbed [here](https://github.com/github/codeql/pull/11742/files#diff-0c75be2b1a02487e04ac197b1e372a3db6ac7b46507adcef91049698add3064fR125-R126).

Comment (atorralba) [reply swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift:209]: Yes, it's a forced cast to keep it simple. That would probably not work if the code was actually executed, but it should be good enough for the test.

**Review by geoffw0 — DISMISSED** [review]
LGTM.

**Comment by atorralba** [comment]
Had to rebase to fix conflicts with the GRDB PR.

**Pull request merged**